pkg/output/table_filter.go

```diff
@@ -0,0 +1,159 @@
+				newRow = append(newRow, "")
+			}
+		}
+		newTableData = append(newTableData, newRow)
+	}
+
+	return newHeaders, newTableData
+}
+
+func RemoveEmptyColumns(headers []string, tableData [][]string) [][]string {
+	if len(tableData) == 0 {
+		return [][]string{headers}
+	}
+
+	// Use FilterColumns with a predicate that keeps non-empty columns
+	newHeaders, newTableData := FilterColumns(headers, tableData, func(columnIdx int) bool {
+		return !CheckIfColumnIsEmpty(columnIdx, tableData)
+	})
+
+	// Combine headers and data as before for backward compatibility
+	var result [][]string
+	result = append(result, newHeaders)
+	result = append(result, newTableData...)
+
+	return result
+}
+
+func FilterColumnsWithConditions(headers []string, tableData [][]string, preserveColumn map[int]bool) ([]string, [][]string) {
+	return FilterColumns(headers, tableData, func(columnIdx int) bool {
+		// Keep column if it's marked to be preserved OR if it has data
+		return preserveColumn[columnIdx] || !CheckIfColumnIsEmpty(columnIdx, tableData)
+	})
+}
+
+// RemoveEmptyColumnsWithSeparators removes empty columns and filters the separators list accordingly
+func RemoveEmptyColumnsWithSeparators(headers []string, tableData [][]string, separators []string) ([][]string, []string) {
+	if len(tableData) == 0 {
+		return [][]string{headers}, separators
+	}
+
+	var newHeaders []string
+	var newSeparators []string
+	var columnsToKeep []int
+
+	for i, header := range headers {
+		if !CheckIfColumnIsEmpty(i, tableData) {
+			newHeaders = append(newHeaders, header)
+			columnsToKeep = append(columnsToKeep, i)
+			if i < len(separators) {
+				newSeparators = append(newSeparators, separators[i])
+			}
+		}
+	}
+
+	var result [][]string
+	result = append(result, newHeaders)
+
+	for _, row := range tableData {
+		var newRow []string
+		for _, col := range columnsToKeep {
+			if col < len(row) {
+				newRow = append(newRow, row[col])
+			} else {
+				newRow = append(newRow, "")
+			}
+		}
+		result = append(result, newRow)
+	}
+
+	return result, newSeparators
+}
+
+func BuildTable(writer *tabwriter.Writer, separator string, tableData [][]string) {
+	// Build separators list from single separator for backward compatibility
+	if len(tableData) == 0 {
+		return
+	}
+
+	maxCols := 0
+	for _, row := range tableData {
+		if len(row) > maxCols {
+			maxCols = len(row)
+		}
+	}
+
+	separators := make([]string, maxCols)
+	for i := range separators {
+		separators[i] = separator
+	}
+
+	BuildTableWithSeparators(writer, separators, tableData)
+}
+
+// BuildTableWithSeparators writes table data using a list of separators
+func BuildTableWithSeparators(writer *tabwriter.Writer, separators []string, tableData [][]string) {
+	var output string
+
+	for rowIdx, row := range tableData {
+		if len(row) == 0 {
+			continue
+		}
+
+		for colIdx, col := range row {
+			output += col
+			// Add separator after column if not the last column
+			if colIdx < len(row)-1 && colIdx < len(separators) {
```

Comment:
This function looks much better, one suggestion is maybe we don't need the first part of this if statement (please test some hidden columns to make sure im right!)

My thinking is that if the separator list and each row are parallel (same number of items in each slice) then there should be no issues using that alone

Reply:
tested and saw some improvements there, the behavior was buggy, so thanks for the suggestion

Reply:
Could you describe what you mean?

Reply:
When there were fewer separators than columns, for example 2 separators for 4 columns, the && condition would fail for columns 2 and 3, causing no separator at all to be added